site/content/entry/citra-progress-report-2018-Q1.md

```diff
@@ -0,0 +1,245 @@
+role in your gameplay experience. So always choose a sever that is nearby (geographically).
+Please use our [support forums](https://community.citra-emu.org/),
+[IRC](http://webchat.freenode.net/?channels=citra) or [Discord](https://citra-emu.org/discord/)
+to get help regarding issues with multiplayer.
+
+_Read more about the multiplayer feature update in [**here**](https://citra-emu.org/entry/announcing-networking-support/)._
+
+## Citra-qt : Hardware Shader Improvements (post GLvtx) by [wwylele](https://github.com/wwylele) and [degasus](https://github.com/degasus)
+
+Citra has come a long way from just being able to render graphics in 2015, to
+actually being able to run games at a playable state. Still, Citra was only fast
+for users who had better hardware. Users with slower processors, suffered from lag
+and stutters during gameplay.
+
+Now, thanks to the combined efforts of [phantom](https://github.com/phanto-m),
+[jroweboy](https://github.com/jroweboy), [wwylele](https://github.com/wwylele),
+[Merrymage](https://github.com/MerryMage) and many more, we now have a near-complete
+hardware renderer with huge improvements. Citra is now able to make full use of host
+GPU and render the graphics by using GPU power. [wwylele](https://github.com/wwylele)
+and [Merrymage](https://github.com/merrymage) took it upon themselves to simplify
+the humongous heap of code into smaller parts, fixing and testing them.
+
+Thanks to [degasus](https://github.com/degasus) from the dolphin-emu community,
+who imparted his valuable knowledge and understanding of GPUs and OpenGL, we were
+able to optimize the _streaming storage buffer_ support added by [phantom](https://github.com/phanto-m).
+This was done entirely by [degasus](https://github.com/degasus) himself. ([here](https://github.com/citra-emu/citra/pull/3504),[here](https://github.com/citra-emu/citra/pull/3666) and
+[here](https://github.com/citra-emu/citra/pull/3711)).
+The old way to upload the data (both vertex and constants) was a simple glBufferData
+call. But keep in mind that the GPU has a huge execution latency, so you can't just
+copy the new content over the existing buffer. So the driver has to choose between
+different ways to deal with this issue, with different drawbacks:
+
+* 1) Wait for the GPU to finish and copy to the destination buffer. Usually the
+fastest for big uploads, fine on common gaming, terrible for emulators.
+* 2) Allocate a new buffer and tell the GPU which new buffer shall be used now.
+You also need to care about freeing the old buffer once it isn't in used any more.
+This tends to a huge CPU overhead and isn't available on partial buffer updates
+(for eg. Lighting LUTs).
+* 3) Copy the data to an internal staging buffer and ask the GPU to copy it again
+to the destination when it is ready. Twice the memory bandwidth, and switching
+between rendering and copy also takes a while for the GPU. This bottlenecks you
```

Comment:
Dunno if I would use "bottlenecks" like that, maybe "This proves to be a significant bottleneck if".